### DISCUSSION TITLE
Rethink import path solving?

### DISCUSSION THREAD

**Discussion by 2-5** [vuejs/language-tools#1587] [Ideas]
It seems there is a fundamental problem with import path solving in Volar. I keep seeing import path issues fixes in release notes, yet new import path bugs keep on surfacing. 

The back and forth between `import * as path from 'path'` and `import { posix as path } from 'path'` suggests that there is not a clear understanding of what the actual problem or solution is.

Unfortunately I have never worked with either Node or VS Code extensions, so I don't know either.

Some suggestion though to fix this once and for all:

1. Import path resolution needs testing, either with unit tests or with a few test projects that are checked after each path resolution change:
- test relative imports `./util.ts`
- test aliased imports `@/util.ts`
- test absolute imports `util.ts`, `/util.ts`
- test `node_module` imports `import { ref } from "vue"`
- test with a few `tsconfig.json` configurations - `"baseUrl": "."`, `"baseUrl": "src"`
- test with VS Code project where the Vue code does not sit in the project root and `typescript.tsdk` is used: `"typescript.tsdk": "./crm/client/node_modules/typescript/lib"`,
- test with VS Code multi root workspaces
- test same project under macOS/Linux/Windows
 
2. Centralize all path solving through a single path solving function, to avoid requiring changes in so many places when something must be fixed. The function probably needs a `resolutionMethod` kind of argument - use TypeScript/Vue resolution vs use generic path resolution (for CSS? not sure)

3. Look at what other extension are doing (Vetur/React/...), maybe there is a standard solution to this problem

**Comment by johnsoncodehk** [comment]
First of all these are very good and effective suggestion, thank you!
It's a bit time consuming, I would like to check it after finish v1.0 works.
